### DISCUSSION TITLE
Timer Reset (Area Presence Hold / configurable Entity)

### DISCUSSION THREAD

**Discussion by hirsts** [jseidl/magic-areas#367] [Ideas / Feature Requests]
I've been using MA for years and LOVE it. It makes HA useful ;-)

One feature that  I would find useful is a way reset timers / timeouts. 

My approach is to have Shelly devices publish the switch state on a seperate MQTT topic and I use this to turn on and off the MA Area Presence Hold Switch for the given area. For example, if when I go into the garage and I know I will be there for an extended period, I hit the power switch which overrides everything and turns on the lights but also turns on the Area Present Hold Switch so MA doesn't turn off the lights if i've been still for too long. Trust me, if you are of out sensor range, half way up a ladder with a heavy box and the lights go out, it can be a bit sketchy. 

The issue I have is that when I leave the room and turn off the light switch, the lights go out but the 10 minute extended timer is still running and so if you reenter the room within the 10 minutes, the lights don't get auto turned back on. 

Ideally what would happen is that when the Area Hold Switch is turned off then any running timers / timeouts should be cancelled. It could be a configurable option and or you might be able to select a different entity that might reset timers.

Happy for thoughts and to discuss.

**Comment by jseidl** [comment]
I get what you're saying. Magic Areas works great when you have sensors that can reliably tell your presence. Motion sensors are OK for initial sensing but not that good for extended / continuous sensing. You could try using mmWave sensors like the [Aqara FP2](https://www.aqara.com/us/product/presence-sensor-fp2/), [Apollo MSR-2](https://apolloautomation.com/products/msr-2) or [EP1](https://shop.everythingsmart.io/en-us/products/everything-presence-one-kit). These take some configuration to set the range right but works flawlessly.

Needless to say, that doesn't mean MA shouldn't help out people with less sensing capabilities either. The presence hold switch, in code, is no different than any other presence entity and I want to avoid hardcoding this into the presence hold switch due how the whole code works and in an effort to make things predictable (in code) .

I'm planning to introduce some service calls at some point and a service call like `magicareas.clear_area` of sorts would work for your case as you could add an automation for when the presence hold switch goes off, call that service.

**Comment by hirsts** [comment]
Hi Jan,

A service call to reset the timer would be great. This would allow me to reset timers using automation when the presence hold switch is turned off. It would be even more ideal if there was an option in MA to select an entity as a trigger that would trigger the service call, therefore prevent me from needing to write and maintain automations.

Another option is to add an option to the current UI to reset timers when hold switch transitions to off.

I have just ordered some of the new version EP1’s and I have high hopes. If they work well, I plan to hardwire install them in all rooms. I currently use the Philips Hue sensors when have been great and I can normally work around the PIR issues and timings by tuning options in MA. The two main issues with these sensors are that 1) the battery’s run down and I’m not organised enough to change them and 2) they don’t see through glass (this is good and bad). Many times I have forgotten to press the hold switch, got in the shower and then had to shout at Alexa to turn the lights back on ☹

Thanks for responding to the issue / feature request. I’d be happy to be a beta tester when you need one.

Cheers,
Stuart



From: Jan Seidl ***@***.***>
Date: Friday, 21 June 2024 at 4:02 PM
To: jseidl/hass-magic_areas ***@***.***>
Cc: Stuart Hirst ***@***.***>, Author ***@***.***>
Subject: Re: [jseidl/hass-magic_areas] Timer Reset (Area Presence Hold / configurable Entity) (Discussion #367)

I get what you're saying. Magic Areas works great when you have sensors that can reliably tell your presence. Motion sensors are OK for initial sensing but not that good for extended / continuous sensing. You could try using mmWave sensors like the Aqara FP2<https://www.aqara.com/us/product/presence-sensor-fp2/>, Apollo MSR-2<https://apolloautomation.com/products/msr-2> or EP1<https://shop.everythingsmart.io/en-us/products/everything-presence-one-kit>. These take some configuration to set the range right but works flawlessly.

Needless to say, that doesn't mean MA shouldn't help out people with less sensing capabilities either. The presence hold switch, in code, is no different than any other presence entity and I want to avoid hardcoding this into the presence hold switch due how the whole code works and in an effort to make things predictable (in code) .

I'm planning to introduce some service calls at some point and a service call like magicareas.clear_area of sorts would work for your case as you could add an automation for when the presence hold switch goes off, call that service.
